### DISCUSSION TITLE
Relative path leads to ModuleNotFound

### DISCUSSION THREAD

**Discussion by scottyeung** [emmett-framework/granian#400] [Q&A]
Hi there, thanks for making such a rapid framework, I thought it could be used as a drop-in replacement for FastAPI but when I tried to run via `root@:/app/app# granian main --workers 8 --port 8001` it just crashed. I had no issues with `Uvicorn` and `socketify` in the past. 

Any guidance will be appreciated. Thanks a bunch!

```
During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.12/multiprocessing/process.py", line 314, in _bootstrap
    self.run()
  File "/usr/local/lib/python3.12/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.12/site-packages/granian/server.py", line 308, in _spawn_rsgi_worker
    target = callback_loader()
             ^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/granian/_internal.py", line 58, in load_target
    module = load_module(path)
             ^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/granian/_internal.py", line 43, in load_module
    raise RuntimeError(
RuntimeError: While importing 'app.app.main', an ImportError was raised:

Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/granian/_internal.py", line 40, in load_module
    __import__(module_name)
  File "/app/app/main.py", line 7, in <module>
    from app.api.api_v1.api import api_router
ModuleNotFoundError: No module named 'app.api'
```

**Comment by gi0baro** [comment]
Can you try to go up one level and target `app.main`? Also, you're missing the `--interface` parameter that has to be set to `asgi`.

**Comment by scottyeung** [comment]
> Can you try to go up one level and target `app.main`? Also, you're missing the `--interface` parameter that has to be set to `asgi`.

Just tried and it's not working still on my Docker environment. Here's the `main.py` I'm working with. How can I resolve the namespace issue if that works locally but not on Docker?

```
from fastapi import FastAPI, Request

from app.api.api_v1.api import api_router
from app.core.config import settings

app = FastAPI(
    title=settings.PROJECT_NAME, openapi_url=f"{settings.API_V1_STR}/openapi.json",
)

# Set up CORS
app.add_middleware(
    CORSMiddleware,
    allow_origins=["*"],  # Allow only the frontend origin
    allow_credentials=True,
    allow_methods=["*"],  # Allows all methods
    allow_headers=["*"],  # Allows all headers
)

app.include_router(api_router, prefix=settings.API_V1_STR)

@app.exception_handler(Exception)
async def global_exception_handler(request: Request, exc: Exception):
    return JSONResponse(
        status_code=500,
        content={"message": "An unexpected error occurred. Please try again later."},
    )
import logging

# Configure logging
logging.basicConfig(level=logging.INFO)
```

**Comment by iamkhav** [comment]
The previous traceback you gave suggests that this import
```python
from app.api.api_v1.api import api_router
```
can not be resolved.

It seems the `app` package is known in your environment but does it have an `api` module (from where you're looking)?
Can you provide your folder structure perhaps?
<sub>^Edit: I removed a suggestion from above that didn't make sense (my bad). Let's look at your folder structure and go from there.

You could isolate this with something like
```python
# some_python_file.py
from app.api.api_v1.api import api_router
print("Hello world.")
```
but frankly, the error you linked doesn't have anything to do with Granian.

You might also want to look at your Dockerfile, especially if you can confirm that things work locally but not in docker?
Maybe you're mounting things in an unintended way?
